File: M2K/pack1/Cluster.java
package pack1;
import java.util.*;
import java.text.*;

public class Cluster{
  private ToaDo centroid;
  private List<ToaDo> pointOfCluster;

  public Cluster(ToaDo center){
    centroid = new ToaDo(center);
    pointOfCluster = new ArrayList<ToaDo>();
  }
  public void add(ToaDo diem){
    pointOfCluster.add(diem);
  }
  public ToaDo center(){
    ToaDo average = new ToaDo(0,0);
    for (ToaDo diem: pointOfCluster){
      average.setX(average.getX() + diem.getX());
      average.setY(average.getY() + diem.getY());
    }
    int pointCount = pointOfCluster.size();
    average.setX(average.getX() / pointCount);
    average.setY(average.getY() / pointCount);
    return average;
  }
  public ToaDo getCentroid(){
    return centroid;
  }
  public List<ToaDo> returnList(){
    return pointOfCluster;
  }
}

File: M2K/pack1/Cluster.py
from pack1.Point import Point

class Cluster:
	def __init__(self, center: Point) -> None:
		self.centroid = Point(center)
		self.pointOfCluster: list[Point] = []
	
	def add(self, diem):
		self.pointOfCluster.append(diem)
	
	def center(self):

		x = 0
		y = 0

		for diem in self.pointOfCluster:
			x += diem.x
			y += diem.y
		
		pointCount = len(self.pointOfCluster)

		x /= pointCount
		y /= pointCount

		return Point(x, y)

	def getCentroid(self):
		return self.centroid
	
	def returnList(self):
		return self.pointOfCluster